src/components/SignIn.js

```diff
@@ -1,49 +1,101 @@
-import React from 'react';
+import React , {useState} from 'react';
 import { Link } from "react-router-dom";
 import picLogScreen from '../assets/picLogScreen.png'
 import { Container, Form, Button, Row, Col} from 'react-bootstrap';
 import './SignIn.css';
+import { axiosInstance } from '../util/config';
+import * as Yup from 'yup';
+import { useNavigate } from "react-router-dom";
 
 export default function SignIn() {
+
+    const [username, setUsername] = useState('');
+    const [password, setPassword] = useState('');
+    const [error, setError] = useState(false); 
+    const [errorMessage, setErrorMessage] = useState('');
+
+    let navigate = useNavigate(); 
+    
+    const signInSchema = Yup.object().shape({
+        username: Yup.string().max(15).required(),
+        password: Yup.string().min(8).required(),
+    });
+
+    const handleClick = async (e) => {
+        e.preventDefault();
+        setError(false); 
+        setErrorMessage(''); 
+        const isValid = await signInSchema.isValid({
+            username: username,
+            password: password,
+        });
+
+        if (isValid) {
+            axiosInstance
+                .post('/auth/signin', {
+                    username: username,
+                    password: password,
+                })
+                .then((response) => {
+                    navigate('/users/:id');
+                    console.log(response);
+                })
+                .catch((err) => {
+                    console.log(err);
+                    console.log(err.response);
+                    setErrorMessage(err.response.data.msg); 
+                    setError(true);
+                });
+        } else {
+            setError(true); 
+            setErrorMessage('One or more input is invalid');
```

Comment:
Sometimes it might not be the input that's the issue, but there might be an issue with the server (i.e. 500 status code)

Reply:
The code above it will catch this problem, this was more for testing purposes.